### PR TITLE
Fix provider compatibility check crash in evp_test

### PR DESCRIPTION
**Pull request by t8m**
EVP_MAC_CTX_get_mac_size() cannot be called on older unfixed versions before EVP_MAC_init().

##### Checklist

- [x] tests are added or updated

### CODE REVIEW

**Review by jamuir — APPROVED** [review]

File: test/evp_test.c
Comment (nhorman) [inline test/evp_test.c:1625]: this is likely just more of a question: is the EVP_MAC code codified in the fips provider?  i.e. if we use an older fips provider with a newer libcrypto, is the fix for EVP_MAC picked up?

Comment (t8m) [reply test/evp_test.c:1625]: No, all crypto (for FIPS approved algorithms) must be inside the FIPS provider. So no, if you use older fips provider with new libcrypto, this bug will still be there, that is the reason we need this condition to actually apply the check.

**Review by nhorman — APPROVED** [review]

**Comment by t8m** [comment]
ping @openssl/committers for second review

**Review by slontis — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master and 3.2 branches. Thank you for the reviews.

**Pull request closed without merging**
